src/lib/scheduler/abstract_task.cpp

```diff
@@ -35,8 +35,11 @@ void AbstractTask::set_as_predecessor_of(std::shared_ptr<AbstractTask> successor
 
   successor->_on_predecessor_added();
   _successors.emplace_back(successor);
+  successor->_predecessors.emplace_back(shared_from_this());
```

Comment:
this and l36 are two calls that update the predecessor info of a task. can you make that one call, e.g., pass shared_from_this() to _on_predecessor_added()?

Reply:
To be honest, I don't really see why we have that `on_predecessor_added` indirection. Looks like abstraction gone to far to me. Personally, I would rather remove it and move the `_pending_predecessors++` here.

Reply:
would be fine with that, too, if on_predecessor_added is only ever called here.